client/state/preferences/reducer.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import has from 'lodash/has';
+import get from 'lodash/get';
+import forOwn from 'lodash/forOwn';
+/**
+ * Internal dependencies
+ */
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+import { USER_SETTING_KEY } from './actions';
+
+function values( state = {}, action ) {
+	if ( action.type === PREFERENCES_RECEIVE && has( action, [ 'data', USER_SETTING_KEY ] ) ) {
+		state = Object.assign( {}, state, get( action, [ 'data', USER_SETTING_KEY ] ) );
+		forOwn( state, ( value, key ) => {
+			if ( null === value ) {
+				delete state[ key ];
+			}
+		} );
+	} else if ( action.type === PREFERENCES_SET && action.key && action.value ) {
+		state = Object.assign( {}, state );
+		state[ action.key ] = action.value;
+	} else if ( action.type === PREFERENCES_REMOVE && action.key ) {
+		state = Object.assign( {}, state );
+		state[ action.key ] = null;
```

Comment:
Right now this is being persisted without safeguards.

If we'd like to persist this, we'll want to add an json schema and validate against that, so our data can change without causing client errors.
```javascript
case DESERIALIZE:
            if ( isValidStateWithSchema( state, itemsSchema ) ) {
                return state;
            }
            return defaultState;
```

If you would not like to persist, return the default state, which is an empty object here for both SERIALIZE and DESERIALIZE. 
```javascript
case DESERIALIZE:
case SERIALIZE:
    return {};
```

See the following for more reasoning and examples https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#data-persistence--2754-